packages/context/src/value-promise.ts

```diff
@@ -191,32 +191,57 @@ export function resolveList<T, V>(
 export function tryWithFinally<T>(
   action: () => ValueOrPromise<T>,
   finalAction: () => void,
+): ValueOrPromise<T> {
+  const errorAction = (err: unknown): never => {
+    throw err;
+  };
+  return tryCatchFinally(action, errorAction, finalAction);
```

Comment:
Can we move the default error action to `tryCatchFinally` so that the implementation of `tryWithFinally` can be simplified as follows?

```suggestion
  return tryCatchFinally(action, undefined, finalAction);
```